### DISCUSSION TITLE
libcurl certificate caching policy

### DISCUSSION THREAD

**Discussion by kasha13** [curl/curl#10761] [Q&A] [answered]
Libcurl does not cache verifying certificates. Investigating libcurl sources

```
cache_criteria_met = (data->set.general_ssl.ca_cache_timeout != 0) &&
           conn_config->verifypeer &&
           !conn_config->CApath &&
           !conn_config->ca_info_blob &&
           !ssl_config->primary.CRLfile &&
           !ssl_config->native_ca_store;
```

cache_criteria_met is false because conn_config->CApath is set to /etc/ssl/certs. All other variables are true. Why CApath is the problem for libcurl to cache? What can I do to enable caching? Libcurl uses openssl 3.

OS: Linux Ubuntu

Compiler: gcc

**Comment by bagder** [comment, marked as answer]
You're more than welcome to improve the code to cache the CA certs even better/more. The current restrictions are in place because that's how the code works. It's just code. It can be changed and improved.

Caching the contents of a *directory* rather than just a single file will of course introduce a whole slew of new challenges, which probably is the reason it works like this right now.
